src/Microsoft.ML.StandardLearners/Standard/LogisticRegression/LbfgsPredictorBase.cs

```diff
@@ -92,77 +93,89 @@ public abstract class ArgumentsBase : LearnerInputBaseWithWeight
-            L2Weight = args.L2Weight;
-            Contracts.CheckUserArg(args.L1Weight >= 0, nameof(args.L1Weight), "Must be non-negative");
-            L1Weight = args.L1Weight;
-            Contracts.CheckUserArg(args.OptTol > 0, nameof(args.OptTol), "Must be positive");
-            OptTol = args.OptTol;
-            Contracts.CheckUserArg(args.MemorySize > 0, nameof(args.MemorySize), "Must be positive");
-            MemorySize = args.MemorySize;
-            Contracts.CheckUserArg(args.MaxIterations > 0, nameof(args.MaxIterations), "Must be positive");
-            MaxIterations = args.MaxIterations;
-            Contracts.CheckUserArg(args.SgdInitializationTolerance >= 0, nameof(args.SgdInitializationTolerance), "Must be non-negative");
-            SgdInitializationTolerance = args.SgdInitializationTolerance;
-            Quiet = args.Quiet;
-            InitWtsDiameter = args.InitWtsDiameter;
-            UseThreads = args.UseThreads;
-            Contracts.CheckUserArg(args.NumThreads == null || args.NumThreads.Value >= 0, nameof(args.NumThreads), "Must be non-negative");
-            NumThreads = args.NumThreads;
-            DenseOptimizer = args.DenseOptimizer;
-            ShowTrainingStats = showTrainingStats;
-            EnforceNonNegativity = args.EnforceNonNegativity;
+            Host.CheckValue(args, nameof(args));
+            Args = args;
+
+            Contracts.CheckUserArg(!Args.UseThreads || Args.NumThreads > 0 || Args.NumThreads == null,
+                nameof(Args.NumThreads), "numThreads must be positive (or empty for default)");
+            Contracts.CheckUserArg(Args.L2Weight >= 0, nameof(Args.L2Weight), "Must be non-negative");
+            Contracts.CheckUserArg(Args.L1Weight >= 0, nameof(Args.L1Weight), "Must be non-negative");
+            Contracts.CheckUserArg(Args.OptTol > 0, nameof(Args.OptTol), "Must be positive");
+            Contracts.CheckUserArg(Args.MemorySize > 0, nameof(Args.MemorySize), "Must be positive");
+            Contracts.CheckUserArg(Args.MaxIterations > 0, nameof(Args.MaxIterations), "Must be positive");
+            Contracts.CheckUserArg(Args.SgdInitializationTolerance >= 0, nameof(Args.SgdInitializationTolerance), "Must be non-negative");
+            Contracts.CheckUserArg(Args.NumThreads == null || Args.NumThreads.Value >= 0, nameof(Args.NumThreads), "Must be non-negative");
+
+            L2Weight = Args.L2Weight;
+            L1Weight = Args.L1Weight;
+            OptTol = Args.OptTol;
+            MemorySize = Args.MemorySize;
+            MaxIterations = Args.MaxIterations;
+            SgdInitializationTolerance = Args.SgdInitializationTolerance;
+            Quiet = Args.Quiet;
+            InitWtsDiameter = Args.InitWtsDiameter;
+            UseThreads = Args.UseThreads;
+            NumThreads = Args.NumThreads;
+            DenseOptimizer = Args.DenseOptimizer;
+            EnforceNonNegativity = Args.EnforceNonNegativity;
```

Comment:
They are been called by both constructors. I set it up so that the first constructor calls the second using this(...). It was suggested by Zeeshan A.

---
In reply to: [219994227](https://github.com/dotnet/machinelearning/pull/957#discussion_r219994227) [](ancestors = 219994227)